src/Tizen.Network.Bluetooth/Tizen.Network.Bluetooth/BluetoothHidDeviceImpl.cs

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) 2016 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Threading.Tasks;
+
+namespace Tizen.Network.Bluetooth
+{
+    internal class BluetoothHidDeviceImpl : IDisposable
+    {
+        private event EventHandler<HidDeviceConnectionStateChangedEventArgs> _hidDeviceConnectionStateChanged;
+        private event EventHandler<HidDeviceDataReceivedEventArgs> _hidDeviceDataReceived;
+
+        private TaskCompletionSource<BluetoothError> _taskForConnection;
+        private bool _taskForConnectionIsCompleted;
+        private TaskCompletionSource<BluetoothError> _taskForDisconnection;
+        private bool _taskForDisconnectionIsCompleted;
+
+        private static readonly BluetoothHidDeviceImpl _instance = new BluetoothHidDeviceImpl();
+        private bool disposed = false;
+
+        internal event EventHandler<HidDeviceConnectionStateChangedEventArgs> ConnectionStateChanged
+        {
+            add
+            {
+                _hidDeviceConnectionStateChanged += value;
+            }
+            remove
+            {
+                _hidDeviceConnectionStateChanged -= value;
+            }
+        }
+
+        internal Task ConnectHidDeviceAsync(string deviceAddress)
+        {
+            _taskForConnection = new TaskCompletionSource<BluetoothError>();
```

Comment:
Please, check previous request was completed, 

``` c#
 if (_taskForConnection != null && !_taskForConneciton.Task.IsCompleted) {
     throw new InvalidOperationException("Connection is in progress");
 }
```

Reply:
Done. Thank you.